File: 01jupyter/nb/source/01_CF_primer/include/fileio.f90
program fileio
  integer :: i, ios
  real(8) :: x
  ! 書き込み: unit 番号を割り当てて開く
  open(unit=10, file="data.txt", status="replace", action="write")
  do i = 0, 4
     write(10, "(i0,1x,f0.6)") i, i * 0.5d0   ! 1行に2つの数を書く
  end do
  close(10)

  ! 読み込み
  open(unit=10, file="data.txt", status="old", action="read")
  do
     read(10, *, iostat=ios) i, x      ! iostat /= 0 で読み終わり
     if (ios /= 0) exit
     print "(a,i0,a,f0.6)", "read: i = ", i, ", x = ", x
  end do
  close(10)
end program fileio

File: 01jupyter/nb/source/01_CF_primer/include/fileio.cpp
#include <cstdio>

int main() {
  /* 書き込み: "w" で開く */
  FILE * wp = fopen("data.txt", "w");
  if (wp == NULL) { printf("cannot open for write\n"); return 1; }
  for (int i = 0; i < 5; i++) {
    fprintf(wp, "%d %f\n", i, i * 0.5);   /* 1行に2つの数を書く */
  }
  fclose(wp);

  /* 読み込み: "r" で開く */
  FILE * rp = fopen("data.txt", "r");
  if (rp == NULL) { printf("cannot open for read\n"); return 1; }
  int i;
  double x;
  /* fscanf が 2 個読めた間繰り返す */
  while (fscanf(rp, "%d %lf", &i, &x) == 2) {
    printf("read: i = %d, x = %f\n", i, x);
  }
  fclose(rp);
  return 0;
}
